core/webgl-compatibility/src/test/Compatibility.test.ts

```diff
@@ -111,63 +114,99 @@ describe("Render Compatibility", () => {
-    const canvas = _createCanvas();
-    expect(canvas).to.not.be.undefined;
-    const context = createContext(canvas!, false);
-    expect(context).to.not.be.undefined;
-
+    const context = makeTestContext();
     const caps = new Capabilities();
-    const compatibility = caps.init(context!, ["EXT_frag_depth"]);
+    const compatibility = caps.init(context, ["EXT_frag_depth"]);
     expect(compatibility.status).to.equal(WebGLRenderCompatibilityStatus.MissingOptionalFeatures);
     expect(compatibility.missingOptionalFeatures.indexOf(WebGLFeature.FragDepth)).to.not.equal(-1);
     expect(caps.supportsFragDepth).to.be.false;
   });
 
   it("should query proper render compatibility info assuming lack of instancing support", () => {
-    const canvas = _createCanvas();
-    expect(canvas).to.not.be.undefined;
-    const context = createContext(canvas!, false);
-    expect(context).to.not.be.undefined;
-
+    const context = makeTestContext();
     const caps = new Capabilities();
-    const compatibility = caps.init(context!, ["ANGLE_instanced_arrays"]);
+    const compatibility = caps.init(context, ["ANGLE_instanced_arrays"]);
     expect(compatibility.status).to.equal(WebGLRenderCompatibilityStatus.MissingOptionalFeatures);
     expect(compatibility.missingOptionalFeatures.indexOf(WebGLFeature.Instancing)).to.not.equal(-1);
   });
 
   it("should query proper render compatibility info assuming lack of standard derivatives support", () => {
-    const canvas = _createCanvas();
-    expect(canvas).to.not.be.undefined;
-    const context = createContext(canvas!, false);
-    expect(context).to.not.be.undefined;
-
+    const context = makeTestContext();
     const caps = new Capabilities();
-    const compatibility = caps.init(context!, ["OES_standard_derivatives"]);
+    const compatibility = caps.init(context, ["OES_standard_derivatives"]);
     expect(compatibility.status).to.equal(WebGLRenderCompatibilityStatus.MissingOptionalFeatures);
     expect(compatibility.missingOptionalFeatures.indexOf(WebGLFeature.StandardDerivatives)).to.not.equal(-1);
   });
+
+  it("detects early Z culling driver bug", () => {
+    const renderers = [
+      [ "ANGLE (Intel(R) HD Graphics 630 Direct3D11 vs_5_0 ps_5_0)", true ],
+      [ "ANGLE (Intel(R) UHD Graphics 630 Direct3D11 vs_5_0 ps_5_0)", true ],
+      [ "ANGLE (Intel(R) HD Graphics 620 Direct3D11 vs_5_0 ps_5_0)", true ],
+      [ "ANGLE (Intel(R) UHD Graphics 620 Direct3D11 vs_5_0 ps_5_0)", true ],
+      [ "ANGLE (Intel HD Graphics 620 Direct3D11 vs_5_0 ps_5_0)", false ],
+
+      // Bug only confirmed on 620 and 630
+      [ "ANGLE (Intel(R) UHD Graphics 610 Direct3D11 vs_5_0 ps_5_0)", false ],
+      [ "ANGLE (Intel(R) UHD Graphics 615 Direct3D11 vs_5_0 ps_5_0)", false ],
+      [ "ANGLE (Intel(R) UHD Graphics 500 Direct3D11 vs_5_0 ps_5_0)", false ],
+      [ "ANGLE (Intel(R) UHD Graphics 520 Direct3D11 vs_5_0 ps_5_0)", false ],
```

Comment:
UHD only applies 600, so may be best to make these two HD

Reply:
[The bug](https://dev.azure.com/bentleycs/beconnect/_workitems/edit/484001) contains a screenshot from the user in which unmasked renderer string contains "UHD Graphics 630".

Reply:
Ah, I see what you're saying. The test is checking to make sure we *don't* detect these as false positives. They don't have to match real renderes. I'll add "HD".

Reply:
Yes, I meant to type UHD only applies to 6xx (at least so far), not 5xx.